java/core/src/main/java/com/whylogs/core/metrics/IntegralMetric.java

```diff
@@ -62,7 +63,14 @@ public OperationResult columnarUpdate(PreprocessedColumn data) {
     return OperationResult.ok(successes);
   }
 
-  // @Override // TODO: move this to a factory so it can be accessed for building
+  @Override
+  public HashMap<String, MetricComponent> getComponents() {
+    HashMap<String, MetricComponent> components = new HashMap<>();
+    components.put(this.maxComponent.getTypeName(), this.maxComponent);
+    components.put(this.minComponent.getTypeName(), this.minComponent);
+    return components;
+  }
+
```

Comment:
Is there a way to make the config argument to `zero()` optional?

Reply:
Yes I can make a second method that allows that. I'll add it in.